build.gradle

```diff
@@ -309,23 +319,36 @@ scalafmt {
     // configFilePath = ".scalafmt.conf" // .scalafmt.conf in the project root is default value, provide only if other location is needed
 }
 
-
 configurations.all {
-    resolutionStrategy {
-        eachDependency { DependencyResolveDetails dependencyResolveDetails ->
-            final requestedDependency = dependencyResolveDetails.requested
-            if (requestedDependency.name != 'beam-utilities') {
-                force 'javax.media:jai_core:1.1.3'
+        resolutionStrategy {
+            eachDependency { DependencyResolveDetails dependencyResolveDetails ->
+                final requestedDependency = dependencyResolveDetails.requested
+                if (requestedDependency.name != 'beam-utilities') {
+                    force 'javax.media:jai_core:1.1.3'
+                }
             }
         }
-    }
-    exclude group: 'javax.media', module: 'jai_codec'
-    exclude group: 'javax.media', module: 'jai_imageio'
-
+        exclude group: 'javax.media', module: 'jai_codec'
+        exclude group: 'javax.media', module: 'jai_imageio'
 }
 
 //compileScala.dependsOn(scalafmtAll)
 
+tasks.withType(ScalaCompile) {
+    def additionalParameters = []
+    if ( project.hasProperty("report") ) {
+        additionalParameters += ["-Xplugin:" + configurations.scalaCompilerPlugin.asPath,
+                              "-P:scapegoat:dataDir:" + buildDir + "/reports/scapegoat",
+                              "-P:scapegoat:reports:html:xml",
+                              "-P:scapegoat:disabledInspections:OptionGet:UnsafeTraversableMethods:ComparingFloatingPointTypes:ComparingUnrelatedTypes:TryGet:EitherGet",
+                              "-P:scapegoat:verbose:false",
+                              "-P:scapegoat:consoleOutput:true"]
+    }
+    scalaCompileOptions.additionalParameters = additionalParameters + ["-unchecked", "-deprecation", "-feature"]
```

Comment:
because of fetal warning build was failing becuase of plugin warning messages

Reply:
Then we need to fix those before we push this in - fatal warnings is necessary imho

Reply:
Revert - fatal warnings.  Also plugin will be run based on condition(with `report` flag in compile command) so build will not fail in normal run. It will only fail if we generate report of scapegoat plugin